### PR TITLE
Fix issues reported for v1.8.1 release

### LINKED ISSUE #313: Release version 1.9.0
This issue is for tracking changes for the 1.9.0 release. Target release date: **November 2024**

## [Release instructions](https://github.com/10up/simple-podcasting/blob/develop/CONTRIBUTING.md#release-instructions)

- [x] Branch: Starting from `develop`, cut a release branch named `release/1.9.0` for your changes.
- [x] Version bump: Bump the version number in `simple-podcasting.php`, `package-lock.json`, `package.json`, and `readme.txt` if it does not already reflect the version being released.  Update both the plugin "Version:" property and the plugin `PODCASTING_VERSION` constant in `simple-podcasting.php`.
- [x] Changelog: Add/update the changelog in both `CHANGELOG.md` and `readme.txt`.
- [x] Props: update `CREDITS.md` with any new contributors, confirm maintainers are accurate.
- [x] New files: Check to be sure any new files/paths that are unnecessary in the production version are included in `.distignore`.
- [x] Readme updates: Make any other readme changes as necessary. `README.md` is geared toward GitHub and `readme.txt` contains WordPress.org-specific content. The two are slightly different.
- [x] Merge: Make a non-fast-forward merge from your release branch to `develop` (or merge the pull request), then do the same for `develop` into `trunk`, ensuring you pull the most recent changes into `develop` first (`git checkout develop && git pull origin develop && git checkout trunk && git merge --no-ff develop`). `trunk` contains the latest stable release.
- [x] Push: Push your `trunk` branch to GitHub (e.g. `git push origin trunk`).
- [x] [Compare](https://github.com/10up/simple-podcasting/compare/trunk...develop) `trunk` to `develop` to ensure no additional changes were missed.
- [x] Test the pre-release ZIP locally by [downloading](https://github.com/10up/simple-podcasting/actions/workflows/build-release-zip.yml) it from the Build release zip action artifact and installing it locally. Ensure this zip has all the files we expect, that it installs and activates correctly and that all basic functionality is working.
- [x] Release: Create a [new release](https://github.com/10up/simple-podcasting/releases/new), naming the tag and the release with the new version number, and targeting the `trunk` branch.  Paste the changelog from `CHANGELOG.md` into the body of the release and include a link to the [closed issues on the milestone](https://github.com/10up/simple-podcasting/milestone/21?closed=1).
- [x] SVN: Wait for the [GitHub Action](https://github.com/10up/simple-podcasting/actions) to finish deploying to the WordPress.org repository. If all goes well, users with SVN commit access for that plugin will receive an emailed diff of changes.
- [x] Check WordPress.org: Ensure that the changes are live on [WordPress.org](https://wordpress.org/plugins/simple-podcasting/). This may take a few minutes.
- [x] Close the milestone: Edit the [milestone](https://github.com/10up/simple-podcasting/milestone/21) with release date (in the `Due date (optional)` field) and link to GitHub release (in the `Description` field), then close the milestone.
- [x] Punt incomplete items: If any open issues or PRs which were milestoned for `1.9.0` do not make it into the release, update their milestone to `2.0.0`or `Future Release`.

### PR DESCRIPTION
**Pull request by sudar**
### Description of the Change

This PR fixes some of the issues reported in #313 

The following are the two issues that are fixed in this PR.

1. In the settings for each Podcast you can set a Podcast cover image. Within each Podcast block, you can also set an episode image. The preview within the editor always shows the Podcast cover image, not the Podcast block episode image. It does show properly on the front-end though.
2. I'm getting a PHP warning on the front-end: Attempt to read property "term_id" on string, pointing to includes/blocks/podcast/markup.php:43

Partially Closes #313 

### How to test the Change

Here are the steps to test the first change

- Create a Podcast show term
- Assign an image to the podcast show
- Create a new post
- Add the Podcast show term to the post
- Add a new Podcast block
- Enable the 'Show Display Art' checkbox
- The Podcast block will show the Podcast show image
- Now add a new episode image
- The preview should change and then show the episode image instead of the podcast show image

Here are the steps to test the second change

- Create a new post
- Don't add any Podcast term to the post
- Add a new Podcast block
- Save the post and view it in the frontend
- There should not be any PHP warnings

### Changelog Entry
- Fixed - Show the preview of the episode image if set at block level
- Fixed - PHP warning when an episode is not associated with any Podcast show

### Credits
Props @sudar 

### Checklist:
- [x] I agree to follow this project's [**Code of Conduct**](https://github.com/10up/.github/blob/trunk/CODE_OF_CONDUCT.md).
- [x] I have updated the documentation accordingly.
- [x] I have added [Critical Flows, Test Cases, and/or End-to-End Tests](https://10up.github.io/Open-Source-Best-Practices/testing/) to cover my change.
- [x] All new and existing tests pass.

### CODE REVIEW

File: assets/js/edit.js [hunk 1]
```diff
@@ -436,7 +436,18 @@ function Edit( props ) {
...
+									<div className="wp-block-podcasting-podcast__image">
+										<img
+											src={featuredImageUrl}
+											alt={showName}
+										/>
+									</div>
+								</div>
+							)}
+
+							{displayArt && !featuredImageUrl && showImage && (
```
Comment (dkotter) [inline assets/js/edit.js]: If possible, would be great to combine all of this in a single conditional, as the markup we render is the same for both and right now we have pretty much duplicate code. Something along the lines of check if `displayArt` is `true` and then within that conditional, determine if we are showing the featured image or podcast image

File: includes/blocks/podcast/markup.php [hunk 2]
```diff
@@ -40,7 +40,11 @@
 $episode_type   = get_post_meta( $post_id, 'podcast_episode_type', true );
 $episode_number = get_post_meta( $post_id, 'podcast_episode_number', true );
 $season_number  = get_post_meta( $post_id, 'podcast_season_number', true );
-$term_image_id  = get_term_meta( $podcast_show->term_id, 'podcasting_image', true );
+if ( is_a( $podcast_show, 'WP_Term' ) ) {
+	$term_image_id  = get_term_meta( $podcast_show->term_id, 'podcasting_image', true );
```
Comment (dkotter) [inline includes/blocks/podcast/markup.php]: Extra space here:
```suggestion
	$term_image_id = get_term_meta( $podcast_show->term_id, 'podcasting_image', true );
```

File: includes/blocks/podcast/markup.php [hunk 3]
```diff
@@ -40,7 +40,11 @@
 $episode_type   = get_post_meta( $post_id, 'podcast_episode_type', true );
 $episode_number = get_post_meta( $post_id, 'podcast_episode_number', true );
 $season_number  = get_post_meta( $post_id, 'podcast_season_number', true );
-$term_image_id  = get_term_meta( $podcast_show->term_id, 'podcasting_image', true );
+if ( is_a( $podcast_show, 'WP_Term' ) ) {
```
Comment (dkotter) [inline includes/blocks/podcast/markup.php:43]: Note that this solves the undefined issue but doesn't actually solve the root problem, where the proper data isn't actually set on any existing blocks, even if the data was there before. I'm fine with adding this defensive check but just want to ensure we don't lose the thread on the larger issue

**Review by dkotter — APPROVED** [review]

**Review by dkotter — CHANGES_REQUESTED** [review]
Unrelated to the requested changes but there was a PHPCS error that wasn't failing the GitHub Action. I've gone ahead and updated that check to use the latest version of our PHPCS library and have fixed all existing PHPCS errors

File: assets/js/edit.js (hunk 1 above)
Comment (iamdharmesh) [reply assets/js/edit.js]: This is handled in https://github.com/10up/simple-podcasting/pull/318/commits/33d39ae6aed0e94a612346c021b0f355443d6d07

**Review by dkotter — APPROVED** [review]

**Pull request merged**